stripped/colabs/wizualizacja_algorytmu_nieomylnego_stripped.py

```diff
@@ -0,0 +1,450 @@
+        niepustynia_przy_CD['Evap'].quantile(0.25),
+        CP['Evap'].quantile(0.75),
+        niepustynia_przy_CP['Evap'].quantile(0.25),
+        GBD['Evap'].quantile(0.75),
+        niepustynia_przy_GBD['Evap'].quantile(0.25),
+    ]).mean()
+)
+
+Evap_graniczne
+
+AvgSurfT_graniczne = round(
+    pd.Series([
+        CD['AvgSurfT'].quantile(0.25),
+        niepustynia_przy_CD['AvgSurfT'].quantile(0.75),
+        CP['AvgSurfT'].quantile(0.25),
+        niepustynia_przy_CP['AvgSurfT'].quantile(0.75),
+        GBD['AvgSurfT'].quantile(0.25),
+        niepustynia_przy_GBD['AvgSurfT'].quantile(0.75),
+    ]).mean()
+)
+
+AvgSurfT_graniczne
+
+Albedo_graniczne = round(
+    pd.Series([
+        CD['Albedo'].quantile(0.25),
+        niepustynia_przy_CD['Albedo'].quantile(0.75),
+        CP['Albedo'].quantile(0.25),
+        niepustynia_przy_CP['Albedo'].quantile(0.75),
+        GBD['Albedo'].quantile(0.25),
+        niepustynia_przy_GBD['Albedo'].quantile(0.75),
+    ]).mean(), 1
+)
+
+Albedo_graniczne
+
+SoilT_40_100cm_graniczne = round(
+    pd.Series([
+        CD['SoilT_40_100cm'].quantile(0.25),
+        niepustynia_przy_CD['SoilT_40_100cm'].quantile(0.75),
+        CP['SoilT_40_100cm'].quantile(0.25),
+        niepustynia_przy_CP['SoilT_40_100cm'].quantile(0.75),
+        GBD['SoilT_40_100cm'].quantile(0.25),
+        niepustynia_przy_GBD['SoilT_40_100cm'].quantile(0.75),
+    ]).mean()
+)
+
+SoilT_40_100cm_graniczne
+
+from pyspark.sql import DataFrame as SparkDataFrame, functions as F
+
+def heuristic_classify_winter(data: SparkDataFrame) -> SparkDataFrame:
+    assert all(
+        [column for column in data.columns if "_condition" not in column]
+    ), "the data must not contain condition in any column name"
+    less_than_columns = {
+        "Rainf": 12,
+        "Evap": 13,
+        "GVEG": 0.211,
+    }
+    greater_than_columns = {
+        "AvgSurfT": 273,
+        "Albedo": 33.4,
+        "SoilT_40_100cm": 275,
+    }
+    minimal_condition_count = 4
+    columns_to_check = list(less_than_columns.keys()) + list(
+        greater_than_columns.keys()
+    )
+    for column, value in less_than_columns.items():
+        data = data.withColumn(
+            f"{column}_condition", F.when(F.col(column) <= value, 1).otherwise(0)
+        )
+    for column, value in greater_than_columns.items():
+        data = data.withColumn(
+            f"{column}_condition", F.when(F.col(column) >= value, 1).otherwise(0)
+        )
+    return (
+        data.withColumn(
+            "conditions_fullfiled_sum",
+            sum(F.col(column + "_condition") for column in columns_to_check),
+        )
+        .withColumn(
+            "Pustynia",
+            F.when(
+                F.col("conditions_fullfiled_sum") >= minimal_condition_count, 1
+            ).otherwise(0),
+        )
+        .drop(
+            "conditions_fullfiled_sum",
+            *[column for column in data.columns if "_condition" in column],
+        )
+    )
```

Comment:
Nie do końca rozumiem po co to, ale chyba powinno być  w src obok mojej funkcji.

Reply:
To nowa funkcja stworzona w oparciu o chłodniejsze miesiące, które nie występowały w początkowej wersji algorytmu. Wartości są wyznaczane we wcześniejszych linijkach. W takim wypadku też powinnam dodać?